### PR TITLE
Remove proxy folder

### LINKED ISSUE #1689: Remove proxy folder
Move everything from proxy folder to the repository root to simplify repo structure.

Part of #1577

### PR DESCRIPTION
**Pull request by DimaStebaev**
- remove proxy folder
- fix github actions workflow
- fix eslint warnings
- add spell check

### CODE REVIEW

**Review by yavrsky — DISMISSED** [review]

File: docs/deployment.md [hunk 1]
```diff
@@ -39,7 +40,7 @@ Deploy process includes:
 6)  Node modules should be initialized in the following folders:
     ```shell
     export IMA_ROOT=.....
-    cd $IMA_ROOT/proxy; npm install
+    cd $IMA_ROOT; npm install
     cd $IMA_ROOT/npms/skale-owasp; npm install
     cd $IMA_ROOT/npms/skale-ima; npm install
     cd $IMA_ROOT/agent; npm install
```
Comment (DmytroNazarenko) [inline docs/deployment.md]: IMA agent dirs could be removed

File: package.json [hunk 2]
```diff
@@ -1,22 +1,64 @@
 {
-    "name": "skale-ima",
+    "name": "skale-ima-proxy",
```
Comment (DmytroNazarenko) [inline package.json:2]: Why should we rename the package?

File: predeployed/test/requirements.txt [hunk 3]
```diff
@@ -1 +1,2 @@
 predeployed-generator>=1.2.0
+setuptools
```
Comment (DmytroNazarenko) [inline predeployed/test/requirements.txt:2]: Why don't you specify the version?

File: package.json (hunk 2 above)
Comment (DimaStebaev) [reply package.json:2]: It was the name of smart contracts package in `proxy` folder.

File: predeployed/test/requirements.txt (hunk 3 above)
Comment (DimaStebaev) [reply predeployed/test/requirements.txt:2]: Any version works. It's needed only for testing environment.

**Review by DmytroNazarenko — APPROVED** [review]

**Pull request merged**